functorch/__init__.py

```diff
@@ -91,7 +91,15 @@ def _functorch_str(tensor):
         return _old_str(tensor)
 
     value = _C.get_unwrapped(tensor)
+    # Disable temporarily kDynamicLayerFrontModeKey/kDynamicLayerBackModeKey as included dispatch keys
+    dl_enabled = _C.tls_set_is_included()
+    if (dl_enabled):
+        _C._set_dynamic_layer_keys_included(False)
     value_repr = repr(value)
+    # Reenable kDynamicLayerFrontModeKey/kDynamicLayerBackModeKey as included dispatch keys
+    if (dl_enabled):
+        _C._set_dynamic_layer_keys_included(True)
+
```

Comment:
Ugh I think I introduced this bug last month. I agree on the approach (that when we're printing the repr we should disable all things functorch related). Let me think about this a little more and get back to you

Reply:
I thought about it and it looks like this was a pre-existing condition (and not caused by my refactor of DynamicLayer last month)